go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -657,20 +657,31 @@ func (qre *QueryExecutor) execDDL(conn *StatefulConnection) (*sqltypes.Result, e
 		}
 	}
 
-	defer func() {
-		// Call se.Reload() with includeStats=false as obtaining table
-		// size stats involves joining `information_schema.tables`,
-		// which can be very costly on systems with a large number of
-		// tables.
-		//
-		// Instead of synchronously recalculating table size stats
-		// after every DDL, let them be outdated until the periodic
-		// schema reload fixes it.
-		if err := qre.tsv.se.ReloadAtEx(qre.ctx, mysql.Position{}, false); err != nil {
-			log.Errorf("failed to reload schema %v", err)
-		}
-	}()
-
+	isTemporaryTable := false
+	switch stmt := qre.plan.FullStmt.(type) {
+	case *sqlparser.CreateTable:
+		isTemporaryTable = stmt.IsTemporary()
+	case *sqlparser.DropTable:
+		isTemporaryTable = stmt.IsTemporary()
+	}
+
+	if !isTemporaryTable {
```

Comment:
In the analyzeDDL plan
we mark in the plan `NeedsReservedConn` to true, so we can use that field here and ignore the `reload`

```go
return &Plan{PlanID: PlanDDL, FullQuery: fullQuery, FullStmt: stmt, NeedsReservedConn: stmt.IsTemporary()}, nil
```

Reply:
Can there be a future scenario where `NeedsReservedConn` is `true` even if the table is not `TEMPORARY`?

Reply:
I've meanwhile simplified the check for `IsTemporary()`

Reply:
I do not expect that to change and I have added a test specifically for that. 
We can keep the current simplified version as well.

Reply:
let's keep the existing version, then, as it is more explicit :pray: